spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/database/support/OraclePagingQueryProvider.java

```diff
@@ -29,7 +29,22 @@
 	@Override
 	public String getPageQuery(Pageable pageable) {
 		int offset = pageable.getOffset()+1;
-		return SqlPagingQueryUtils.generateRowNumSqlQueryWithNesting(this, getSelectClause(), getSelectClause(), false, "TMP_ROW_NUM >= "
+		return generateRowNumSqlQueryWithNesting(getSelectClause(), getSelectClause(), false, "TMP_ROW_NUM >= "
 				+ offset + " AND TMP_ROW_NUM < " + (offset+pageable.getPageSize()));
 	}
+
+	private String generateRowNumSqlQueryWithNesting(String innerSelectClause,
```

Comment:
Since this is a private method and you pass in the same for both inner and outer, why not just pass it in once?